app/routes/chatbot.js

```diff
@@ -112,74 +114,93 @@ router.post('/', (req, res) => {
+                return reject(err);
+              }
+
+              return resolve(campaign);
+            });
         }
 
-        const userDeclined = !req.incoming_message || !helpers.isYesResponse(req.incoming_message);
-        if (userDeclined) {
-          // Feels a little hacky to throw an error to break chain, but it's simple enough to catch.
-          throw new Error('broadcast declined');
+        if (scope.keyword) {
+          logger.debug(`load campaign for keyword:${scope.keyword}`);
+          campaignId = app.locals.keywords[scope.keyword];
+          campaign = app.locals.campaigns[campaignId];
+
+          if (!campaign) {
+            const msg = `Campaign not found for keyword '${scope.keyword}'.`;
+            const err = new NotFoundError(msg);
+            return reject(err);
+          }
+
+          if (campaign.status === 'closed') {
+            // Store campaign to render in closed message.
+            scope.campaign = campaign;
+            // TODO: Include this message to the CampaignClosedError.
+            const msg = `Keyword received for closed campaign ${campaignId}.`;
+            const err = new UnprocessibleEntityError(msg);
+            return reject(err);
+          }
+
+          return resolve(campaign);
         }
 
-        return campaign;
-      }
-
-      if (scope.keyword) {
-        logger.debug(`load campaign for keyword:${scope.keyword}`);
-        campaignId = app.locals.keywords[scope.keyword];
-        campaign = app.locals.campaigns[campaignId];
+        campaign = app.locals.campaigns[user.current_campaign];
+        logger.debug(`user.current_campaign:${user.current_campaign}`);
 
         if (!campaign) {
-          const msg = `Campaign not found for keyword '${scope.keyword}'.`;
-          throw new NotFoundError(msg);
+          // TODO: Send to non-existent start menu to select a campaign.
+          const msg = `User ${user._id} current_campaign ${campaignId} not found in CampaignBot.`;
+          const err = new NotFoundError(msg);
+          return reject(err);
         }
 
-        if (campaign.status === 'closed') {
-          // Store campaign to render in closed message.
-          scope.campaign = campaign;
-          // TODO: Include this message to the CampaignClosedError.
-          const msg = `Keyword received for closed campaign ${campaignId}.`;
-          throw new UnprocessibleEntityError(msg);
-        }
-
-        return campaign;
-      }
-
-      campaignId = user.current_campaign;
-      campaign = app.locals.campaigns[campaignId];
-      logger.debug(`user.current_campaign:${campaignId}`);
-
-      if (!campaign) {
-        // TODO: Send to non-existent start menu to select a campaign.
-        const msg = `User ${user._id} current_campaign ${campaignId} not found in CampaignBot.`;
-        throw new NotFoundError(msg);
-      }
+        return resolve(campaign);
+      });
+  });
 
-      return campaign;
-    })
+  return loadCampaign
```

Comment:
Adding a semi-colon where would break the promise chain below. Would it be better if this line read as 
`return loadCampaign.then((campaign) => {`? The other promises are using the same syntax (where the `then` is on a new line).